server/repository/src/db_diesel/name_store_join.rs

```diff
@@ -45,13 +45,20 @@ pub struct NameStoreJoinRow {
     pub name_is_supplier: bool,
 }
 
+#[derive(PartialEq, Debug, Clone, Default)]
+pub struct NameStoreJoin {
+    pub name_store_join: NameStoreJoinRow,
+    pub name: NameRow,
+}
+
 joinable!(name_store_join -> store (store_id));
 joinable!(name_store_join -> name_link (name_link_id));
 
-type NameStoreJoin = (NameStoreJoinRow, (NameLinkRow, NameRow));
+type NameStoreJoins = (NameStoreJoinRow, (NameLinkRow, NameRow));
```

Comment:
Slighty confusing names but I don't have a better suggestion...

Reply:
Haha yea a funny one. NameStoreJoinJoin?? Honestly I've felt it a little bit weird having that `XxXxJoin` naming throughout the repo layer. Could these types be more broadly named `QueryResult`?

Reply:
Though I usually have refrain from using `Result` in any naming if we're not dealing with the rust `Result` type!

Reply:
We could name it NameStoreJoinDiesel because its the diesel related query type. But I think just keeping it as it is is probably easier/better.

Reply:
DieselOutsieWowsie